### PR TITLE
pano_logic_g2: Set RTL serial port baudrate from environment BAUD if …

### PR DESCRIPTION
**Pull request by skiphansen**
…present.

### CODE REVIEW

File: targets/pano_logic_g2/base.py [hunk 1]
```diff
@@ -26,6 +27,10 @@ class BaseSoC(SoCSDRAM):
     def __init__(self, platform, **kwargs):
         dict_set_max(kwargs, 'integrated_rom_size', 0x8000)
         dict_set_max(kwargs, 'integrated_sram_size', 0x8000)
+        if os.environ.get('BAUD', '0') == '0':
+            kwargs['uart_baudrate']=19200
+        else:
+            kwargs['uart_baudrate']=int(os.environ.get('BAUD'))
```
Comment (mithro) [inline targets/pano_logic_g2/base.py:33]: Maybe something like the following;
```python
        baud = os.environ.get('BAUD', '19200')
        try:
            kwargs['uart_baudrate'] = int(baud)
        except TypeError as e:
            raise ArgumentError('BAUD value in environment {} is not a valid number!'.format(baud)
```

I also wonder if this should be part of the `make.py` script to enable it for all designs?

Comment (mithro) [reply targets/pano_logic_g2/base.py:33]: I guess it should probably check that there isn't a `uart_baudrate` already in kwargs?

**Comment by mithro** [comment]
@skiphansen - I think the idea of this change makes sense and I support adding it -- just a couple of random comments around the implementation.

File: targets/pano_logic_g2/base.py (hunk 1 above)
Comment (skiphansen) [reply targets/pano_logic_g2/base.py:33]: makes sense to me.  I was trying to limit my damage to Pano.  I'll give that a try.  Thanks for the review!

**Comment by skiphansen** [comment]
I'll be submitting a new patch that's more general.

**Pull request closed without merging**